p2p/libp2p/netMessenger.go

```diff
@@ -467,20 +498,20 @@ func (netMes *networkMessenger) RegisterMessageProcessor(topic string, handler p
 		return err
 	}
 
-	netMes.topics[topic] = handler
+	netMes.processors[topic] = handler
 	return nil
 }
 
 // UnregisterMessageProcessor registers a message processes on a topic
 func (netMes *networkMessenger) UnregisterMessageProcessor(topic string) error {
 	netMes.mutTopics.Lock()
 	defer netMes.mutTopics.Unlock()
-	validator, found := netMes.topics[topic]
 
+	_, found := netMes.topics[topic]
 	if !found {
 		return p2p.ErrNilTopic
 	}
-
+	validator := netMes.processors[topic]
```

Comment:
validator, ok := ... if !ok

Reply:
actually, this variant covers the case in witch (due to a programming error) we inserted value nil with a topic key. In that case ok would have returned true but the value would have been nil